.github/actions/conventional-checks/action.yml

```diff
@@ -0,0 +1,49 @@
+name: Conventional Checks
+
+description: Check if the branch name and pull request title follow conventional checks.
+
+runs:
+  using: composite
+  steps:
+    - name: Get branch name (merge)
+      if: github.event_name != 'pull_request'
+      shell: bash
+      run: echo "BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_ENV
+
+    - name: Get branch name (pull request)
+      if: github.event_name == 'pull_request'
+      shell: bash
+      run: echo "BRANCH_NAME=$(echo ${GITHUB_HEAD_REF})" >> $GITHUB_ENV
+
+    - name: Check branch name
+      shell: bash
+      run: |
+        local_branch="${{ env.BRANCH_NAME }}"
+
+        valid_branch_regex="^(improvement|bug|feature|test|tmp)\/AAE-[0-9]+-[A-Za-z0-9._-]+$"
+
+        message="Branch name $local_branch must adhere to this contract: $valid_branch_regex."
+        message="$message Your commit will be rejected. You should rename your branch to a valid name like "
```

Comment:
strictly speaking the commit cannot be rejected, I will try to rephrase with something like:

```
The branch name doesn't follow the expected conventions
```

Reply:
I would be in favor or a "directive" message telling people directly what to do. Something along the lines of 

```
The branch name doesn't follow the expected conventions: rename your branch following the regexp [insert regexp from input], close this pull request, and open a new one.
```

Also instead of `message="$message ..."` I prefer `message+=" ..."` as it's more concise, wdyt?

Reply:
I would also add `please` before `rename your branch`